src/Arch/Microchip/Common/PICRegisterUniqueAddress.cs

```diff
@@ -0,0 +1,106 @@
+﻿#region License
+/* 
+ * Copyright (C) 2017-2018 Christian Hostelet.
+ * inspired by work from:
+ * Copyright (C) 1999-2018 John Källén.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Reko.Arch.MicrochipPIC.Common
+{
+    [DebuggerDisplay("{_debugDisplay,nq}")]
+    public class PICRegisterUniqueAddress :
+        IComparable<PICRegisterUniqueAddress>, IComparer<PICRegisterUniqueAddress>,
+        IEquatable<PICRegisterUniqueAddress>, IEqualityComparer<PICRegisterUniqueAddress>,
+        IComparable
+    {
```

Comment:
Need a comment here to explain the purpose of this class.

Reply:
Explanation added. To be discussed if clear enough.